server/src/main/java/org/opensearch/index/translog/transfer/FileSnapshot.java

```diff
@@ -67,10 +69,13 @@ public long getContentLength() throws IOException {
     public InputStream inputStream() throws IOException {
         return fileChannel != null
             ? new BufferedChecksumStreamInput(
-                new InputStreamStreamInput(Channels.newInputStream(fileChannel), fileChannel.size()),
-                path.toString()
+                new InputStreamStreamInput(new BufferedInputStream(Channels.newInputStream(fileChannel))),
+                name
             )
-            : new BufferedChecksumStreamInput(new BytesStreamInput(content), name);
+            : new BufferedChecksumStreamInput(
+                new InputStreamStreamInput(new InputStreamIndexInput(new ByteArrayIndexInput(this.name, content), content.length)),
+                name
```

Comment:
What does `BufferedChecksumStreamInput` do? Does it add checksum integrity while uploading and reading the data? Since we are making changes in the upload path, should there be any corresponding change in download path as well?

Reply:
> Since we are not using checksums while uploading to remote .

Yeah, I think the plan is to use checksum while uploading as well as downloading. But I got your point. This can be added as we start passing the custom checksum to the configured remote store.

> Does it add checksum integrity while uploading and reading the data?

Yes, while uploading. For downloading, `BufferedChecksumStreamOutput` will be used.

> Since we are making changes in the upload path, should there be any corresponding change in download path as well?

We are not making changes to upload path regarding checksum. `BufferedChecksumStreamInput` was already present. As part of this change, we are passing InputStream that supports `mark` and `reset` to `BufferedChecksumStreamInput`.

Reply:
For now, I will make the changes as @gbbafna suggested.